docs/pages/reference/backends.mdx

```diff
@@ -172,6 +172,143 @@ teleport:
+[`wal_level`](https://www.postgresql.org/docs/current/runtime-config-wal.html#GUC-WAL-LEVEL)
+parameter must be set to `logical` and
+[`max_replication_slots`](https://www.postgresql.org/docs/current/runtime-config-replication.html#GUC-MAX-REPLICATION-SLOTS)
+must be set to at least as many Teleport auth servers as you'll be running (a
+higher number is recommended, to account for network timeouts and
+reconnections). In addition, the database user must have the `initiating
+replication` [role
+attribute](https://www.postgresql.org/docs/current/role-attributes.html) (as
+well as `login privilege`, of course). Since replication permissions allow for
+essentially full read access over the entire cluster (with a physical
+replication connection) or to all databases that the user can connect to, it's
+recommended to prevent the user from opening replication connections, or from
+connecting to databases other than the ones used for Teleport, if the PostgreSQL
+cluster is shared between Teleport and other applications.
+
+To configure Teleport to use PostgreSQL:
+- Configure all Teleport Auth servers to use the PostgreSQL back-end in the
+  `storage` section of `teleport.yaml` as shown below.
+- Deploy several auth servers connected to the PostgreSQL storage back-end.
+- Deploy several proxy nodes.
+- Make sure that the proxy instances and all Teleport resource services that
+  connect directly to the the auth service have the `auth_server` configuration
+  setting populated with the address of a load balancer for auth service
+  instances.
+
+```yaml
+teleport:
+  storage:
+    type: postgresql
+
+    # conn_string is a libpq-compatible connection string (see
+    # https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING);
+    # pool_max_conns is an additional parameter that determines the maximum
+    # number of connections in the connection pool used for the cluster state
+    # database (the change feed uses an additional connection), defaulting to
+    # a value that depends on the number of available CPUs.
+    conn_string: postgresql://user_name@database-address/teleport_backend?sslmode=verify-full&pool_max_conns=20
+
+    # An audit_events_uri with a scheme of postgresql:// will use the
+    # PostgreSQL backend for audit log storage; the URI is a libpq-compatible
+    # connection string just like the cluster state conn_string, but cannot be
+    # specified as key=value pairs. It's possible to specify completely
+    # different PostgreSQL clusters for cluster state and audit log.
+    audit_events_uri:
+      - postgresql://user_name@database-address/teleport_audit?sslmode=verify-full
+```
+
+Audit log events are periodically deleted after a default retention period of
+3766 hours (one year); it's possible to select a different retention period or
+to disable the cleanup entirely, by specifying the `retention_period` or the
+`disable_cleanup` parameters in the fragment of the URI:
```

Comment:
Why are these in the fragment and not just additional query params?

Reply:
I didn't want to accidentally interfere with some other parameter that one might want to set in the connection string itself; this way we have a distinction between Teleport parameters and database connection parameters.